controllers/nstemplateset/client_test.go

```diff
@@ -0,0 +1,199 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"k8s.io/client-go/kubernetes/scheme"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+func TestApplyToolchainObjects(t *testing.T) {
+	// given
+	role := newRole("john-dev", "edit-john", "john")
+	devNs := newNamespace("advanced", "john", "dev")
+	dBaaSTenant := newDBaaSTenant("john-dev-tenant", "john")
+	additionalLabel := map[string]string{
+		"foo": "bar",
+	}
+
+	t.Run("when create two", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(role, devNs), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, changed)
+		assertObject(t, fakeClient, false)
+	})
+
+	t.Run("when create only one, the second is present", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+		_, err := client.NewApplyClient(fakeClient, scheme.Scheme).Apply(objects(role), additionalLabel)
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(role, devNs), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, changed)
+		assertObject(t, fakeClient, false)
+	})
+
+	t.Run("when only DBaaSTenant is supposed to be applied but the group for DBaaS is not present", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+		_, err := client.NewApplyClient(fakeClient, scheme.Scheme).Apply(objects(role, devNs), additionalLabel)
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(dBaaSTenant), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.False(t, changed)
+		assertObject(t, fakeClient, false)
+	})
+
+	t.Run("when only DBaaSTenant is supposed to be applied and the group for DBaaS is present", func(t *testing.T) {
+		// given
+		apiClient, fakeClient := prepareAPIClient(t)
+		apiClient.AvailableAPIGroups = append(apiClient.AvailableAPIGroups, metav1.APIGroup{Name: "dbaas.redhat.com"})
+		_, err := client.NewApplyClient(fakeClient, scheme.Scheme).Apply(objects(role, devNs), additionalLabel)
+		require.NoError(t, err)
+
+		// when
+		changed, err := apiClient.ApplyToolchainObjects(logger, objects(dBaaSTenant), additionalLabel)
+
+		// then
+		require.NoError(t, err)
+		assert.True(t, changed)
+		assertObject(t, fakeClient, true)
+	})
+}
+
+func objects(objects ...runtimeclient.Object) []runtimeclient.Object {
+	var objs []runtimeclient.Object
+	for i := range objects {
+		objs = append(objs, objects[i].DeepCopyObject().(runtimeclient.Object))
+	}
+	return objs
+}
+
+func assertObject(t *testing.T, client *test.FakeClient, expectDBaaSTenant bool) {
```

Comment:
should be named `assertObjects` since it verifies the `Role`, the `Namespace` and optionally the `DBaaSTenant`

Reply:
renamed 6c01621